src/System.Text.RegularExpressions/ref/System.Text.RegularExpressions.cs

```diff
@@ -281,4 +297,11 @@ public abstract partial class RegexRunnerFactory
         protected RegexRunnerFactory() { }
         protected internal abstract System.Text.RegularExpressions.RegexRunner CreateInstance();
     }
+    public ref struct SplitEnumerator
```

Comment:
This should be nested inside Regex.

Reply:
that seems strange to me. Just that I understand, what's the guidance here?

Reply:
> what's the guidance here?

The guidance is that Enumerator structs should be nested inside the type that returns them, e.g.
https://source.dot.net/#System.Private.CoreLib/shared/System/Collections/Generic/Dictionary.cs,1079
https://source.dot.net/#System.Private.CoreLib/shared/System/Collections/Generic/List.cs,1173
https://source.dot.net/#System.Collections.Immutable/System/Collections/Immutable/ImmutableList_1.Enumerator.cs,27

Developers import many different namespaces into their code files, and so we try to avoid duplicating type names throughout the framework, even when they're in different namespaces.  "SplitEnumerator" is too generic a name to poach; making it a nested type addresses that.

Reply:
got it. thanks 👍